### ISSUE TITLE
Potions duplicate on hit

### ISSUE THREAD

**Issue by AmProsius** [AmProsius/gothic-1-community-patch#239]
**Describe the bug**
When the player character drinks a potion and is hit while the animation plays, the effect of the potion is applied, but the potion is not consumed.

**Expected behavior**
When the player character drinks a potion and is hit while the animation plays, the potion will now correctly be consumed.

**Additional context**
Bug provided by [Blubbler](https://forum.worldofplayers.de/forum/threads/1574630-Release-Gothic-1-Community-Patch/page2?p=26716794&viewfull=1#post26716794).

**Comment by szapp** [comment]
It seems that there might not be an easy way to fix it. I have not looked at this at all, but I think the on_state function (where any consumable effects are triggered) is called on animation onset while the item is only removed when the animation played to a specific frame.

Now changing the time of triggering the on_state function to when the animation has finished would lead to the opposite, much more severe problem: The consumable is gone but the effect was not applied. This may be unfortunate for one-time bonuses, but very bad for health increase: In a fight with quickly attacking enemies the player could never restock the health because the animation would always be interrupted.

An alternative which might be impossible would be to tie the triggering of the on_state function to the frame of when the item is removed. Whether this is possible I don't know.

**Comment by pawbuj1981** [comment]
I believe every game has some glitch like this mentioned above.  I wouldn't touch sucha a things, it may leads to further problems. 

BTW there is lack of progress from last 17 days. Is everything fine with a project ?

**Comment by AmProsius** [comment]
> BTW there is lack of progress from last 17 days. Is everything fine with a project ?

Yes, everything's fine. The project pauses for private reasons at the moment, but will continue in the third quarter of 2021.

**Comment by pawbuj1981** [comment]
Please check new Union zBugFixes, PackStringFix which fixes msome inventory bugs like item disseaperance.
